File: code_blocks/integrations/third-party-integrations/amplitude_1.ts
import Purchases from '@revenuecat/purchases-react-native';
import { init, setUserId, getDeviceId } from '@amplitude/analytics-react-native';

async function setup() {
  await Purchases.configure({ apiKey: 'public_sdk_key', appUserID: 'my_app_user_id' });
  await init('amplitude_api_key');
  setUserId('my_app_user_id');
  const deviceId = await getDeviceId();
  await Purchases.setAttributes({ '$amplitudeDeviceId': deviceId });
}

File: code_blocks/integrations/third-party-integrations/amplitude_1.dart
import 'package:purchases_flutter/purchases_flutter.dart';
import 'package:amplitude_flutter/amplitude.dart';

Future<void> setup() async {
  await Purchases.configure(
    PurchasesConfiguration('public_sdk_key')..appUserId = 'my_app_user_id',
  );
  final amplitude = Amplitude.getInstance(instanceName: 'default');
  await amplitude.init('amplitude_api_key', 'my_app_user_id');
  final deviceId = await amplitude.getDeviceId();
  await Purchases.setAttributes({'\$amplitudeDeviceId': deviceId});
}
